### PR TITLE
EVG-8017 Add debugability to shell.exec

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: agent/command/shell.go [hunk 1]
```diff
@@ -69,6 +72,11 @@ type shellExec struct {
 	// allows following commands to execute even if this shell command fails.
 	ContinueOnError bool `mapstructure:"continue_on_err"`
 
+	// Debug is a temporary and deliberately undocumented feature added in EVG-8017
+	// and should be removed. It writes a file called debug_shell_exec.<id>.yml that
```
Comment (ablack12) [inline agent/command/shell.go:76]: Are we ticketing this removal? Does STM only need this for some time?

File: agent/command/shell.go [hunk 2]
```diff
@@ -69,6 +72,11 @@ type shellExec struct {
...
+	// contains the contents of the command and its expansions.
+	Debug bool `map:"debug"`
```
Comment (ablack12) [inline agent/command/shell.go:78]: Does this tag need mapstructure as opposed to map?

File: agent/command/shell.go (hunk 1 above)
Comment (bsamek) [reply agent/command/shell.go:76]: Opened https://jira.mongodb.org/browse/EVG-14311

File: agent/command/shell.go (hunk 2 above)
Comment (bsamek) [reply agent/command/shell.go]: Ha, thanks, fixed. Weirdly I tested this end-to-end in staging, and it worked.

Comment (ablack12) [reply agent/command/shell.go]: Yeah for this the default mapstructure tag would still be `debug`, but

**Comment by bsamek** [comment]
evergreen merge

**Pull request merged**